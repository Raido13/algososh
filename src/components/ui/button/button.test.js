import { Button } from "./button";
import { render } from '@testing-library/react';

describe('Компонент Button', () => {
  test('Отрисовка кнопки: Без текста', () => {
    const {asFragment} = render(<Button />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка кнопки: С текстом', () => {
    const {asFragment} = render(<Button text="Текст" />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });
  
  test('Отрисовка кнопки: Отключена', () => {
    const {asFragment} = render(<Button disabled={true} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка кнопки: Лоадер', () => {
    const {asFragment} = render(<Button isLoader={true} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка кнопки: Клик', () => {
    const {asFragment} = render(<Button onClick={() => jest.fn()} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  })
})