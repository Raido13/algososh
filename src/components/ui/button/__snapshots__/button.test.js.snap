// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Button Отрисовка кнопки: Без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Компонент Button Отрисовка кнопки: Клик 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Компонент Button Отрисовка кнопки: Лоадер 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Компонент Button Отрисовка кнопки: Отключена 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Компонент Button Отрисовка кнопки: С текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Текст
  </p>
</button>
`;
